Microsoft.DotNet.Interactive.PowerShell/Microsoft.DotNet.Interactive.PowerShell.csproj

```diff
@@ -27,59 +27,66 @@
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
   </ItemGroup>
 
   <!-- Get NuGetPackageRoot if it's not defined -->
-  <Target
-    Name="GetNuGetPackageRoot"
-    Condition=" '$(NuGetPackageRoot)' == '' "
-  >
+  <Target Name="GetNuGetPackageRoot" Condition=" '$(NuGetPackageRoot)' == '' ">
     <Exec Command="dotnet nuget locals global-packages --list --force-english-output" ConsoleToMSBuild="true">
       <Output TaskParameter="ConsoleOutput" PropertyName="OutputOfExec" />
     </Exec>
 
     <PropertyGroup>
       <RegexForOutputOfExec>info : global-packages: (.*)</RegexForOutputOfExec>
-      <RegexMatch>$([System.Text.RegularExpressions.Regex]::Match($(OutputOfExec), $(RegexForOutputOfExec)).Groups[2].Value)</RegexMatch>
+      <RegexMatch>$([System.Text.RegularExpressions.Regex]::Match($(OutputOfExec), $(RegexForOutputOfExec)).Groups[1].Value)</RegexMatch>
       <NuGetPackageRoot>$(RegexMatch)</NuGetPackageRoot>
     </PropertyGroup>
   </Target>
 
-  <!-- Copy restored contents of PowerShell module nuget pacakges into output -->
-  <ItemGroup>
-    <Content
-      Include="$(NuGetPackageRoot)packagemanagement\$(PackageManagementVersion)\**"
-      Exclude="$(NuGetPackageRoot)**\*.nupkg;$(NuGetPackageRoot)**\*.sha512"
-      PackageCopyToOutput="true">
-      <Link>Modules\PackageManagement\%(RecursiveDir)\%(FileName)%(Extension)</Link>
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </Content>
-    <Content
-      Include="$(NuGetPackageRoot)powershellget\$(PowerShellGetVersion)\**"
-      Exclude="$(NuGetPackageRoot)**\*.nupkg;$(NuGetPackageRoot)**\*.sha512"
-      PackageCopyToOutput="true">
-      <Link>Modules\PowerShellGet\%(RecursiveDir)\%(FileName)%(Extension)</Link>
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </Content>
-    <Content
-      Include="$(NuGetPackageRoot)microsoft.powershell.archive\$(MicrosoftPowerShellArchiveVersion)\**"
-      Exclude="$(NuGetPackageRoot)**\*.nupkg;$(NuGetPackageRoot)**\*.sha512"
-      PackageCopyToOutput="true">
-      <Link>Modules\Microsoft.PowerShell.Archive\%(RecursiveDir)\%(FileName)%(Extension)</Link>
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </Content>
-    <Content
-      Include="$(NuGetPackageRoot)threadjob\$(ThreadJobVersion)\**"
-      Exclude="$(NuGetPackageRoot)**\*.nupkg;$(NuGetPackageRoot)**\*.sha512"
-      PackageCopyToOutput="true">
-      <Link>Modules\ThreadJob\%(RecursiveDir)\%(FileName)%(Extension)</Link>
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </Content>
-  </ItemGroup>
+  <Target Name="CopyGalleryModules" AfterTargets="Restore;CollectPackageReferences">
+    <!-- PackageManagement -->
+    <CreateItem Include="$(NuGetPackageRoot)packagemanagement\$(PackageManagementVersion)\**"
+                Exclude="$(NuGetPackageRoot)**\*.nupkg;$(NuGetPackageRoot)**\*.nuspec;$(NuGetPackageRoot)**\*.sha512;$(NuGetPackageRoot)**\fullclr\**">
```

Comment:
If we exclude the fullclr folder, what will happen to the catalog file? It will be wrong, right? And if the execution policy is ever honored (or forced due to GPO) that could get us in a bad state.

Reply:
I think it's fine. The `PackageManagement` module shipped with PowerShell 7.0 (and all previous powershell releases) only has the coreclr folder. We filter out the fullclr folder during our build.

Reply:
The catalog file is not used when importing the module. I guess It's only used by PowerShellGet or PS Gallery.

Reply:
Sounds good!